open_city_profile/graphene.py

```diff
@@ -178,3 +182,35 @@ def __init_subclass_with_meta__(
             _meta=_meta,
             **options,
         )
+
+
+class AllowedDataFieldsMiddleware:
+
+    def resolve(self, next, root, info, **kwargs):
+
+        operation_type = info.operation.operation
+        if operation_type == OperationType.QUERY and getattr(
+            root, "check_allowed_data_fields", False
+        ):
```

Comment:
We probably should include all operation types (i.e. we shouldn't check if the operation is query). Checking only queries allows the service to access the fields through doing a mutation and querying every field though that.

```suggestion
        if getattr(root, "check_allowed_data_fields", False):
```

Reply:
Yes we should, again very good observation!